source/custom_messages.cpp

```diff
@@ -1145,7 +1146,14 @@ static std::map<std::string, int> pixelWidthTable = {
 // Calculates the maximum number of characters that could satisfy
 // next line in the string from the last calculated newline.
 static size_t NextLineLength(std::string* textStr, size_t lastNewline) {
-    constexpr size_t maxLinePixelWidth = 287;
+    size_t maxLinePixelWidth = 287;
+    size_t lastCaret         = textStr->rfind('^', lastNewline);
+    size_t lastItemIcon      = textStr->rfind("\x7F\x0F", lastNewline);
+
+    // For text boxes with an item icon, decrease max line pixel width to leave space for the icon
+    if (lastItemIcon != std::string::npos && (lastCaret == std::string::npos || lastItemIcon > lastCaret)) {
+        maxLinePixelWidth = 252;
```

Comment:
Normally the text starts 56 pixels from the left of the screen, but with an icon it starts 104 pixels from the left instead, so 48 pixels further in. Any reason you reduced it by 35 pixels instead?

Reply:
I tried some different values and converged on one that fixed all the overflowing texts while changing the least amount of already correct texts.
Reducing the length more would add more newlines where they're not needed, causing more conflicts with the manually inserted ones.